File: API/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
         public int Rating { get; set; }
        public string Description { get; set; }
        public long Price { get; set; }
        public string PictureUrl { get; set; }
        public string New_Arrival { get; set; }
        public string TypeofProduct { get; set; }
        public string Brand { get; set; }
        public int QuantityInStock { get; set; }
    }
}

File: client/src/app/models/product.ts
export interface Product
{
        id: number;
        name: string;
        rating?: number;
        description: string;
        price: number;
        pictureUrl: string;
        new_Arrival?: string;
        typeofProduct?: string;
        brand: string;
        quantityInStock?: number;
    }

    export interface ProductParams {
        orderBy : string,
        searchTerm? : string, //this is optional
        pageNumber : number,
        pageSize : number,
        brandList? : string[], //this is optional
        typeList? : string[] //this is optional
    }